miden-crypto/src/ecdh/k256.rs

```diff
@@ -38,11 +40,26 @@ impl SharedSecret {
     ///
     /// This basically converts a shared secret into uniformly random values that are appropriate
     /// for use as key material.
-    pub fn extract<D>(&self, salt: Option<&[u8]>) -> Hkdf<Sha256, SimpleHmac<Sha256>> {
+    pub fn extract(&self, salt: Option<&[u8]>) -> Hkdf<Sha256, SimpleHmac<Sha256>> {
         self.inner.extract(salt)
     }
 }
 
+impl AsRef<[u8]> for SharedSecret {
+    fn as_ref(&self) -> &[u8] {
+        self.inner.raw_secret_bytes()
+    }
+}
+
+impl Zeroize for SharedSecret {
+    fn zeroize(&mut self) {
+        let zero_bytes = FieldBytes::default();
+        let dummy_secret: k256::ecdh::SharedSecret = zero_bytes.into();
+
+        self.inner = dummy_secret;
+    }
+}
```

Comment:
This seems slightly off, since the compiler might optimize this away. Can't we call `self.inner.zeroize()`?